### DISCUSSION TITLE
Version 0.10 Alpha

### DISCUSSION THREAD

**Discussion by riclolsen** [riclolsen/json-scada#29] [General]
* Default login credentials: username=admin password=jsonscada
* Enhanced MQTT/Sparkplug-B Client driver.
* Fixed commands cancelled from unrelated connections for DNP3, IEC101/104 client drivers.
* Nodejs updated to 16.3.0.
* Nginx update to 1.21.0.
* Telegraf updated to 1.18.3.
* PostgreSQL updated to 13.3.2.
* TimescaleDB updated to 2.2.1.
* MongoDB Community Server updated to 4.4.6.
* MongoDB Compass updated to 1.26.1.
* Chromium updated to 91.0.4472.101.
* Grafana updated to 8.0.1.
* DotNet runtime updated to 5.0.7.
 
Requirements:

* Windows 10 64 bits or Server 2016, Windows PowerShell.

<hr /><em>This discussion was created from the release <a href='https://github.com/riclolsen/json-scada/releases/tag/V0.10-alpha'>Version 0.10 Alpha</a>.</em>

**Comment by joyshmitz** [comment]
Congratulations on 0.10 alpha,
it is a milestone in the development of a great product.
Can you please tell Ricardo how to update correctly from version 8?

**Reply by riclolsen** [reply]
Thanks. I hope to reach 1.0 later this year.

In-place updates are not safe mostly because of database data files. I will try making updates of minor versions possible after 1.0.

Before any attempt to update, it is important to stop all services and make a full backup copy of the current c:\json-scada folder so that you can return to the previous version in case you have any problem. SVGs and any other files you changed should be also backed up.

The safest way to update is to fully dump/backup mongo and postgresql databases (check mongo and postgresql docs on how to do this), uninstall the old version (this will erase everything), install the new version and restore the database files. Restore also your SVGs and other files from the full backup.

It is also possible to keep current local mongo/postgresql binaries and data. In this case, you can copy the c:\json-scada\bin\ and c:\json-scada\src\ folders from the installation of the new version on another computer. For grafana-runtime you can copy all binary folders (keeping the old data folder) from a new install to the current. Or drop all files from a downloaded new version of Grafana to the grafana-runtime folder. Old dashboards are preserved in data/grafana.db file.

Best regards,
Ricardo

**Reply by joyshmitz** [reply]
Thanks for the advice, I'll try to update in the lab
